client-tools.md

```diff
@@ -44,9 +44,9 @@ the WLCG Authorization Working Group's requirements for a command line tool that
 
 | Tool | Status | Comment |
 | -- | -- | --- |
-| oidc-agent | https://github.com/indigo-dc/oidc-agent | Active deployment. Does not support restricted audience. Additional password required. |
+| oidc-agent | https://github.com/indigo-dc/oidc-agent | <ul><li>Supports restricted audiences.</li><li>Supports external password stores to mitigate the additional password requirement. </ul>|
```

Comment:
Marcus: What's an external password store, and how does that work?

Reply:
The `--pw-cmd` option to `oidc-add` allows specifying a call to an external tool. It needs to return a password on stdout.

Reply:
Ok, that's flexible, but how does that work in practice to mitigate the additional password requirement?  I could see using a keychain if oidc-agent is run on a laptop or desktop, but I don't see how it could help on a shared submit host.

Reply:
If access to the password store is authorised via kerberos, this would happen via a passwordless ssh to the submit host, wouldn't it?

Reply:
Ok so if kerberos is in use then yes that would be a passwordless ssh to the submit host.  From there the easiest way to access data with forwarded kerberos credentials would be if you have a kerberos-authenticated filesystem that's also encrypted.   I'm not sure they're so common however, and if you had that, why would you just store a password in it as opposed to storing all the credentials in it without a passphrase?  I'm not aware of a separate command line tool you can use to encrypt/decrypt some data with kerberos, are you?  I know it could be done with a vault server, if your pw-cmd would first get a vault token with kerberos and then read data stored encrypted in vault with it.  But if you had that, again it doesn't make a lot of sense to just store a password in it as opposed to all the oidc credentials or as opposed to storing the refresh token in it instead of in oidc-agent.  So I'm still struggling to see a practical use for this.

Reply:
I thought that would be a straightforward way to use oidc-agent passwordless in a kerberos environment.

Basically, we had this discussion in february, when some people used the additional password as an argument not to use oidc-agent. I didn't interfere much back then, but (since from the above we agree) for the record: oidc-agent can be used securely in a passwordless mode, especially in kerberos environments.

Reply:
I'm confused.  How exactly are you suggesting to use it passwordless in kerberos environments?

Reply:
`ssh <submit-host> "oidc-add --pw-cmd \"<some vault command to retrieve the password>\""`

Reply:
Oh, so you mean to use it with vault and a custom script (it's more than one vault command).  I don't consider that to be straightforward.  If you're going through the trouble of setting up a vault server and installing the vault client, then as I said, I don't see why one wouldn't use vault to do oidc authentication and store the client-id, client secret, and refresh token instead of oidc-agent.

Anyway for this document I suggest we say "Requires additional password at startup time, which could be stored in an external password store such as a kerberos-enabled vault server."

Reply:
Ok;  I've updated that in the document.